Library/Homebrew/formula.rb

```diff
@@ -1853,12 +1853,21 @@ def self.formulae_with_cask_dependents(casks)
          .flat_map { |f| [f, *f.runtime_formula_dependencies].compact }
   end
 
-  # An array of all installed {Formula} without {Formula} dependents
+  # An array of all installed {Formula} without runtime {Formula}
+  # dependents for bottles and without build {Formula} dependents
+  # for those built from source.
   # @private
   def self.formulae_with_no_formula_dependents(formulae)
```

Comment:
Is this method still useful/used after being removed from `leave`?

Reply:
Nope, it might be worth creating another file for all of these methods. They are tangentially related to `Formula` but now they're getting very specific.

Reply:
I don't think it makes much sense to merge them into `Cleanup` though.

Reply:
I ended up moving that logic into `Utils::Autoremove`.

Reply:
For my attempt at this I got very confused about these functions. They generally consider runtime & cask dependencies, but are named like `no_dependents`. So making an alternate version that also considered build dependencies would end up looking like `no_dependents_including_build_dependents`. Then you could think of recommended etc. dependencies and need 6 versions including `no_dependents_at_all_i_mean_it`. 😵 

Anyway, I'm glad you figured out the logic and a saner organization of it to boot. 🍻